File: HotDesk_PRAI/Backup/Library/Library.SQLServer/Connection.vb
Imports System.Data.SqlClient

''' <summary>
''' Handler the Oracle Connection Class
''' -------------------------------------------
''' C.C.Yeon    25 April 2011   Initial Version
''' </summary>
''' <remarks></remarks>
Public MustInherit Class Connection
    Implements IDisposable

    Protected _con As SqlConnection
    Protected _cmd As SqlCommand
    Protected _rdr As SqlDataReader
    Protected _tran As SqlTransaction
    Protected _sqladp As SqlDataAdapter

    Private _constr As String = String.Empty
    Public Property ConnectionString() As String
        Get
            Return _constr
        End Get
        Set(ByVal value As String)
            _constr = value
        End Set
    End Property

    Public Sub New(ByVal ConnectionStringName As String)
        Me.ConnectionString = System.Configuration.ConfigurationManager.ConnectionStrings(ConnectionStringName).ToString

        If ConnectionString = String.Empty Then
            Throw New Exception("Invalid Connection String Name That Set At Web Config")
        End If

        Me._con = New SqlConnection(Me.ConnectionString)
        Me._con.Open()
        Me._cmd = _con.CreateCommand
        Me._tran = Me._con.BeginTransaction
        Me._cmd.Transaction = Me._tran
    End Sub

    Public ReadOnly Property Status() As String
        Get
            If _con IsNot Nothing Then
                Return _con.State.ToString
            Else
                Return String.Empty
            End If
        End Get
    End Property

    ''' <summary>
    ''' Commit all the transaction
    ''' </summary>
    Public Sub Commit()
        Me._tran.Commit()
    End Sub

    ''' <summary>
    ''' Rollback all the transaction
    ''' </summary>
    Public Sub Rollback()
        Me._tran.Rollback()
    End Sub

    Private disposedValue As Boolean = False        ' To detect redundant calls
    ' IDisposable
    Protected Overridable Sub Dispose(ByVal disposing As Boolean)
        If Not Me.disposedValue Then
            If disposing Then
                ' TODO: free other state (managed objects).
            End If

            ' TODO: free your own state (unmanaged objects).
            ' TODO: set large fields to null.

            If _rdr IsNot Nothing Then
                _rdr.Dispose()
            End If

            If _cmd IsNot Nothing Then
                _cmd.Dispose()
            End If

            If _con IsNot Nothing Then
                If _con.State = ConnectionState.Open Then
                    _con.Close()
                End If

                _con.Dispose()
            End If
        End If
        Me.disposedValue = True
    End Sub

#Region " IDisposable Support "
    ' This code added by Visual Basic to correctly implement the disposable pattern.
    Public Sub Dispose() Implements IDisposable.Dispose
        ' Do not change this code.  Put cleanup code in Dispose(ByVal disposing As Boolean) above.
        Dispose(True)
        GC.SuppressFinalize(Me)
    End Sub
#End Region
End Class

File: HotDesk_SPICE/Library/Library.SQLServer/Connection.cs
using System.Data.SqlClient;

namespace Library.SQLServer
{
    /// <summary>
    /// Handler the SQL Server Connection Class
    /// -------------------------------------------
    /// C.C.Yeon    25 April 2011   Initial Version
    /// </summary>
    public abstract class Connection : System.IDisposable
    {
        protected SqlConnection _con;
        protected SqlCommand _cmd;
        protected SqlDataReader _rdr;
        protected SqlTransaction _tran;
        protected SqlDataAdapter _sqladp;

        private string _constr = string.Empty;
        public string ConnectionString
        {
            get { return _constr; }
            set { _constr = value; }
        }

        public Connection(string connectionStringName)
        {
            this.ConnectionString = System.Configuration.ConfigurationManager.ConnectionStrings[connectionStringName].ToString();

            if (ConnectionString == string.Empty)
            {
                throw new System.Exception("Invalid Connection String Name That Set At Web Config");
            }

            this._con = new SqlConnection(this.ConnectionString);
            this._con.Open();
            this._cmd = _con.CreateCommand();
            this._tran = this._con.BeginTransaction();
            this._cmd.Transaction = this._tran;
        }

        public string Status
        {
            get
            {
                if (_con != null)
                {
                    return _con.State.ToString();
                }
                else
                {
                    return string.Empty;
                }
            }
        }

        /// <summary>
        /// Commit all the transaction
        /// </summary>
        public void Commit()
        {
            this._tran.Commit();
        }

        /// <summary>
        /// Rollback all the transaction
        /// </summary>
        public void Rollback()
        {
            this._tran.Rollback();
        }

        private bool disposedValue = false;

        protected virtual void Dispose(bool disposing)
        {
            if (!this.disposedValue)
            {
                if (disposing)
                {
                    // TODO: free other state (managed objects).
                }

                // TODO: free your own state (unmanaged objects).
                // TODO: set large fields to null.

                if (_rdr != null)
                {
                    _rdr.Dispose();
                }

                if (_cmd != null)
                {
                    _cmd.Dispose();
                }

                if (_con != null)
                {
                    if (_con.State == System.Data.ConnectionState.Open)
                    {
                        _con.Close();
                    }
                    _con.Dispose();
                }
            }
            this.disposedValue = true;
        }

        #region IDisposable Support
        public void Dispose()
        {
            Dispose(true);
            System.GC.SuppressFinalize(this);
        }
        #endregion
    }
}
